app/assets/javascripts/button_confirmation.js.coffee

```diff
@@ -20,6 +20,11 @@ $ ->
   apply_button_confirmation()
 
 @apply_button_confirmation = () ->
+  # Remove popover when clicked outside. Reference: https://jsfiddle.net/guya/fjZja/
```

Comment:
That's some dark magic!  I worry that this might be brittle since it depends on a specific implementation of popover which might change.  I think it is okay though since this is an enhancement which wouldn't hinder the user if it randomly breaks.